### PR TITLE
libct: fix two CPU mask nits from #5343

### LINKED ISSUE #5388: CPU reset mask excludes `configs.MaxCPU`
`configs.MaxCPU` is the highest CPU ID runc accepts, but `tryResetCPUAffinity` currently creates its mask with:

```go
unix.NewCPUSet(configs.MaxCPU)
```

`NewCPUSet` takes an exclusive upper bound, so this creates a mask for CPU IDs `0..MaxCPU-1` and omits `MaxCPU` itself.

The reset mask should include every CPU ID runc accepts, so the allocation should be:

```go
unix.NewCPUSet(configs.MaxCPU + 1)
```

This is an off-by-one at the upper boundary.

### PR DESCRIPTION
**Pull request by kolyshkin**
Two follow-up fixes to #5343, found while re-reading the CPU mask changes.

## libct: use len() to check for empty CPU masks

When `CPUAffinity.Initial`/`Final` and `LinuxMemoryPolicy.Nodes` were `*unix.CPUSet`, a nil check was a complete "unset" test. As `unix.CPUSetDynamic` (a slice) it no longer is — a non-nil zero-length slice is now representable, and `{"Initial":[]}` in `state.json` unmarshals to exactly that.

Such a value passes the `!= nil` check and reaches `sched_setaffinity(2)` as a NULL pointer with a zero size, which the kernel reads as an empty mask and rejects with `EINVAL`, instead of resetting the affinity as intended.

Only reachable via a hand-edited `state.json` or a libcontainer user setting the field directly, but `len()` is the idiomatic slice check anyway and treats nil and empty alike. Also drops the now-redundant nil guards in the memory policy validator, since `Count()` on a nil slice is already 0.

`setupMemoryPolicy` is deliberately left as is: an empty `Nodes` is legitimate there, as `MPOL_DEFAULT`/`MPOL_LOCAL` want a NULL/zero-size nodemask, and the validator already rejects an empty `Nodes` for the modes that require one.

## libct: make MaxCPU exclusive

`MaxCPU` was documented and enforced as an inclusive bound (the highest ID `ToCPUSet` accepts), while `unix.NewCPUSet` takes an exclusive one. So `tryResetCPUAffinity`'s `NewCPUSet(MaxCPU)` mask was one ID short of what `ToCPUSet` can parse.

Harmless in practice — `sched_setaffinity(2)` truncates the mask to `cpumask_size()`, and no kernel is configured with anywhere near 64K CPUs — but one constant meaning two different things in two files is asking for a real off-by-one later.

Making `MaxCPU` exclusive means `NewCPUSet(MaxCPU)` is correct as written and `ToCPUSet`'s `NewCPUSet(maxID+1)` can never exceed it. This restores the `ret >= max` check and `max-1` wording that predated the constant, and shifts the test boundary cases down by one.

Fixes: #5388
Closes: #5389

### CODE REVIEW

**Comment by kolyshkin** [comment]
@lifubang @rata PTAL (this fixes real bugs and need to be in before 1.6

**Comment by kolyshkin** [comment]
@opencontainers/runc-maintainers this fixes a real bug, PTAL

**Review by lifubang — APPROVED** [review]

File: libcontainer/configs/validate/validator.go [hunk 1]
```diff
@@ -491,12 +491,12 @@ func memoryPolicy(config *configs.Config) error {
 	}
 	switch mpol.Mode {
 	case unix.MPOL_DEFAULT, unix.MPOL_LOCAL:
-		if mpol.Nodes != nil && mpol.Nodes.Count() != 0 {
```
Comment (AkihiroSuda) [inline libcontainer/configs/validate/validator.go:494]: This nil check was harmless?

**Review by AkihiroSuda — APPROVED** [review]

File: libcontainer/configs/validate/validator.go (hunk 1 above)
Comment (lifubang) [reply libcontainer/configs/validate/validator.go:494]: Yes, it was harmless, just redundant. `Count()` on a nil `CPUSetDynamic` returns 0 (it's a value receiver doing range over the slice), so the explicit nil check adds nothing.
